### PR TITLE
Disable anonymous access and insecure port

### PR DESCRIPTION
**Pull request by jgsqware**
Regarding k8s CIS benchmark, we need to disable anonymous access to the api and use 0 as insecure port.

Currently, k8s-addons use kubectl with anonymous access. We have to provide a kubeconfig for this. eg. proxy-kubeconfig

in k8scc the insecure port is set by `--insecure_port={{.Cluster.Kubernetes.API.InsecurePort}}`

**question:** is it a problem for us to disable it? and do you prefer to remove insecure port from struct and use `0` directly in k8scc or having `0` as default value in the struct and let the operator overriding it

Rel: https://github.com/giantswarm/giantswarm/issues/2020

### CODE REVIEW

**Comment by xh3b4sd** [comment]
I am not sure about the implications of this. Lets wait for an SREs opinion.

**Comment by calvix** [comment]
`k8s-addons` is one of the tools that uses it. I cant recall any other but fore sure this should be a change that is verified by creating a cluster and then running sonobuy k8s conformance test

**Comment by calvix** [comment]
I dont folow this:

> and do you prefer to remove insecure port from struct and use 0 directly in k8scc or having 0 as default value in the struct and let the operator overriding it

 what is the difference

**Comment by calvix** [comment]
I will be personaly sad since i have been using the insecure port for quick access to k8s, but thats just my bad :)

**Comment by xh3b4sd** [comment]
For on prem I am using the insecure port as well to create a SSH tunnel to forward the port to be able to use kubectl locally. How can we do this in case we close the insecure port?

**Comment by jgsqware** [comment]
For on prem, `gsctl` is not working? because I've tested it for AWS and that's working.

@calvix regarding the question, I asking if we disable it at all by default via `k8scloudconfig` or if we let the `x-operator` to set it to 0 by default but use a variable that could be overwritten, eg for testing env...

**Comment by calvix** [comment]
@jgsqware  Thanks for the clarification. 

`gsctl` is working for me for all on-prems and if necessary all k8s apis are accessible from VPN. ( the only problem can be the strange container dance some ppl do on Anubis, but that is not my case as i don't use it).

**Comment by calvix** [comment]
@jgsqware when this is going to be merged can you please make an announcement (`#news-ops`) that the insecure port is disabled on new installations so the ppl are not confused?

**Comment by jgsqware** [comment]
Currently blocked by https://github.com/giantswarm/giantswarm/issues/2062

File: v_0_1_0/master_template.go [hunk 1]
```diff
@@ -1545,7 +1549,7 @@ write_files:
       for manifest in $MANIFESTS
       do
           while
-              /usr/bin/docker run --net=host --rm -v /srv:/srv $KUBECTL apply -f /srv/$manifest
+              /usr/bin/docker run -e KUBECONFIG=${KUBECONFIG} --net=host --rm -v /srv:/srv /etc/kubernetes:/etc/kubernetes $KUBECTL apply -f /srv/$manifest
```
Comment (calvix) [inline v_0_1_0/master_template.go]: you missed `-v` before the `/etc/kubernetes` volume on each line

File: v_0_1_0/master_template.go [hunk 2]
```diff
@@ -1457,21 +1457,27 @@ write_files:
...
 
       /usr/bin/docker pull $KUBECTL
 
+##
+-e KUBECONFIG=${KUBECONFIG} --net=host --rm -v /etc/kubernetes:/etc/kubernetes 
+##
+
       # wait for healthy master
-      while [ "$(/usr/bin/docker run --net=host --rm $KUBECTL get cs | grep Healthy | wc -l)" -ne "3" ]; do sleep 1 && echo 'Waiting for healthy k8s'; done
+      while [ "$(/usr/bin/docker run --e KUBECONFIG=${KUBECONFIG} -net=host --rm /etc/kubernetes:/etc/kubernetes $KUBECTL get cs | grep Healthy | wc -l)" -ne "3" ]; do sleep 1 && echo 'Waiting for healthy k8s'; done
```
Comment (r7vme) [inline v_0_1_0/master_template.go]: remove one dash `--e`

Comment (calvix) [reply v_0_1_0/master_template.go]: also `-net` needs two dashs

File: v_0_1_0/master_template.go (hunk 1 above)
Comment (jgsqware) [reply v_0_1_0/master_template.go]: done

File: v_0_1_0/master_template.go (hunk 2 above)
Comment (jgsqware) [reply v_0_1_0/master_template.go]: done

**Pull request merged**